### PR TITLE
private registry support for DCOS

### PR DESCRIPTION
**Pull request by xtophs**
**What this PR does / why we need it**:
Customer have been asking to add support for deploying from private registries

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

Fixes issue [1136](https://github.com/Azure/acs-engine/issues/1136) 

**Special notes for your reviewer**:

**Release note**:

```release-note
```

---
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/1137)

### CODE REVIEW

**Comment by jchauncey** [comment]
@xtophs could you rebase this?

**Comment by jchauncey** [comment]
The unit tests are failing for this - 

```
Failed to compile acsengine:

# github.com/Azure/acs-engine/pkg/acsengine
pkg/acsengine/engine_test.go:224: undefined: getStorageAccountType
pkg/acsengine/engine_test.go:234: undefined: getStorageAccountType
pkg/acsengine/engine_test.go:244: undefined: getStorageAccountType
```

**Comment by xtophs** [comment]
Remerged, re-tested, rebased 

make test Passes
containers deploy from private registry.

**Comment by jackfrancis** [comment]
@xtophs Thanks for your patience on this. I've run some DCOS tests against this PR and things look good. 2 more things:

- another rebase pretty please (ugh)
- could you kindly move the new config values to be properties of `DcosConfig` instead of `OrchestratorProfile`? (I'll point this out more explicitly in a github review)

File: pkg/api/types.go [hunk 1]
```diff
@@ -137,6 +137,9 @@ type OrchestratorProfile struct {
 	OrchestratorRelease string            `json:"orchestratorRelease"`
 	OrchestratorVersion string            `json:"orchestratorVersion"`
 	KubernetesConfig    *KubernetesConfig `json:"kubernetesConfig,omitempty"`
+	Registry            string            `json:"registry,omitempty"`
```
Comment (jackfrancis) [inline pkg/api/types.go]: Let's move these properties under `DcosConfig`. Documentation, example, and all code usage patterns will need to be adjusted accordingly. Thanks!

**Comment by xtophs** [comment]
@jackfrancis addresses refactoring comment and rebased. Could  you take a look?

**Comment by xtophs** [comment]
bump @jackfrancis ;)

**Comment by jackfrancis** [comment]
@xtophs Thanks for maintaining this! Running tests now.

**Pull request merged**